Cleanup and restructure code and fix bug

enwi opened the pull request:
This also fixes a lot of compiler errors I got when trying to compile with visual studio code and Arduino plugin.

For me these code changes also have the following effect on binary sizes:
Before:
```
Executable segment sizes:
ICACHE : 32768           - flash instruction cache 
IROM   : 364784          - code in flash         (default or ICACHE_FLASH_ATTR) 
IRAM   : 29373   / 32768 - code in IRAM          (IRAM_ATTR, ISRs...) 
DATA   : 1580  )         - initialized variables (global, static) in RAM/HEAP 
RODATA : 2736  ) / 81920 - constants             (global, static) in RAM/HEAP 
BSS    : 31944 )         - zeroed variables      (global, static) in RAM/HEAP 
```

After refactoring:
```
Executable segment sizes:
ICACHE : 32768           - flash instruction cache
IROM   : 364768          - code in flash         (default or ICACHE_FLASH_ATTR)
IRAM   : 29373   / 32768 - code in IRAM          (IRAM_ATTR, ISRs...)
DATA   : 1580  )         - initialized variables (global, static) in RAM/HEAP
RODATA : 2748  ) / 81920 - constants             (global, static) in RAM/HEAP
BSS    : 31928 )         - zeroed variables      (global, static) in RAM/HEAP
```

Hypfer commented on src/esp8266-vindriktning-particle-sensor.ino:
```diff
@@ -1,40 +1,32 @@
 
-unsigned long statusPublishPreviousMillis = millis();
-const long statusPublishInterval = 30000;
+uint16_t statusPublishPreviousMillis = millis();
```
Are you sure about this change from unsigned long to uint16_t?

enwi replied:
Yes an `unsigned long` is in fact a `uint16_t` just look at the definition of `uint16_t`. It is just more platform independent

Hypfer replied:
https://github.com/Hypfer/esp8266-vindriktning-particle-sensor/pull/9/files#diff-2370d1ec15b96074eb253becf564b9bf6e1782204527454069c099c805e5d718R138-R139

Then this should also be uint16_t?

Hypfer replied:
Are you absolutely sure that long is 16 bit on the esp8266?
Because I'm pretty sure it's 32 bit

at least according to
https://www.arduinoforum.de/arduino-Thread-L%C3%A4nge-der-Datentypen-beim-ESP8266

enwi replied:
Also as a side note why are you setting it to `millis()` in the first place? I mean `millis()` is 32 bit, but I know when this is executed it won`t overflow, but setting to 0 should also be fine.

enwi replied:
Technically long long should be 32 bit

enwi replied:
> Are you absolutely sure that long is 16 bit on the esp8266?
> Because I'm pretty sure it's 32 bit
> 
> at least according to
> https://www.arduinoforum.de/arduino-Thread-L%C3%A4nge-der-Datentypen-beim-ESP8266

You are right, it should be uint32_t

Hypfer commented on .clang-format:
```diff
@@ -0,0 +1,58 @@
+AllowShortFunctionsOnASingleLine: Inline
+AlwaysBreakTemplateDeclarations: true
+BreakBeforeBraces: Custom
+BraceWrapping:
```
Please use 1TBS

enwi replied:
1TBS?

Hypfer replied:
https://en.wikipedia.org/wiki/Indentation_style#Variant:_1TBS_(OTBS)

enwi replied:
I was confused, cause I didn't know 1TBS was OTBS and could not find it in the docs

Hypfer commented on src/SerialCom.h:
```diff
@@ -0,0 +1,108 @@
+    uint8_t serialRxBuf[255];
+    uint8_t rxBufIdx = 0;
+
+    void setup() { sensorSerial.begin(9600); }
```
This is missing two newlines

enwi replied:
This was done by clang-format

Hypfer commented on src/SerialCom.h:
```diff
@@ -0,0 +1,108 @@
+
+    bool parseState(particleSensorState_t& state)
+    {
+        uint16_t pm25 = (serialRxBuf[5] << 8) | serialRxBuf[6];
```
While this bitshifting is neat, it is unfortunately much harder to read and understand.

enwi replied:
I can revert this if you want

Hypfer replied:
Actually you're right.
This just triggered the "bitshifts are bad" heuristic.

Hm. To prevent that, maybe a comment above would be a good addition

Something like:
```
/**

                  DF 3     DF 4
      uint16_t  xxxxxxxx xxxxxxxx

*/
```

Of course this is pretty much self-explainatory for people that are skilled in embedded and/or C, however it doesn't hurt to be even more accessible :)

enwi replied:
```c++
        /**
         *         MSB   DF 3    DF 4  LSB
         * uint16_t = xxxxxxxx xxxxxxxx
         */
```

Hypfer commented on src/SerialCom.h:
```diff
@@ -0,0 +1,108 @@
+        {
+            parseState(state);
+
+            Serial.printf("Current measurements: %d, %d, %d, %d, %d\n", state.measurements[0], state.measurements[1],
```
In such cases, I'd use one line per argument as that is much easier to read

enwi replied:
You want it reverted?

Hypfer commented on src/esp8266-vindriktning-particle-sensor.ino:
```diff
@@ -47,228 +39,247 @@ char MQTT_TOPIC_COMMAND[128];
+    snprintf(MQTT_TOPIC_STATE, 127, "%s/%s/state", FIRMWARE_PREFIX, identifier);
+    snprintf(MQTT_TOPIC_COMMAND, 127, "%s/%s/command", FIRMWARE_PREFIX, identifier);
 
+    snprintf(
```
Weird formatting

enwi replied:
This was done by clang-format

Hypfer replied:
I thought so.

Hypfer commented on src/esp8266-vindriktning-particle-sensor.ino:
```diff
@@ -47,228 +39,247 @@ char MQTT_TOPIC_COMMAND[128];
+    for (uint8_t attempt = 0; attempt < 3; ++attempt)
+    {
+        if (mqttClient.connect(
+                identifier, Config::username, Config::password, MQTT_TOPIC_AVAILABILITY, 1, true, AVAILABILITY_OFFLINE))
```
Weird formatting

enwi replied:
This was done by clang-format

Hypfer commented on src/esp8266-vindriktning-particle-sensor.ino:
```diff
@@ -47,228 +39,247 @@ char MQTT_TOPIC_COMMAND[128];
-  delay(3000);
+void setup()
+{
+    Serial.begin(115200);
```
Why 115200?

enwi replied:
Because that's what most people use with ESP`s

Hypfer reviewed (changes requested):
Neat!

enwi commented:
@Hypfer I added a few more changes. Changed the print of the historical values, changed the uint8_t arrays back to chat ones (otherwise MQTT data won't be sent) and also fixed a bug (which was present before), because the function parseState needed to return a boolean, which it didn't so I changed it to a void function. Now everything works really well 😄

enwi commented:
Ohh also do you want to bump the `sw_version`? @Hypfer

Hypfer commented on src/Config.h:
```diff
@@ -0,0 +1,51 @@
+    char password[24] = "";
+
+    void save()
+    {
```
There's still this newline before every function brace

enwi replied:
Need to tweak clang-format

Hypfer commented on src/SerialCom.h:
```diff
@@ -0,0 +1,97 @@
+        if (serialRxBuf[0] == 0x16 && serialRxBuf[1] == 0x11 && serialRxBuf[2] == 0x0B) {
+            parseState(state);
+
+            Serial.printf("Current measurements:");
```
Uh what I ment by the previous comment was more something like

```
    Serial.printf(
      "abc %d %d %d",

      var1,
      var2,
      ...varN
    );
```

enwi replied:
But clang-format won't do that

Hypfer replied:
That's why I don't use auto formatting tools :)

enwi replied:
I just changed it and saved that part without formatting, so that would be fine. Though anyone using the auto formatting would need to do this

Hypfer replied:
In that case please just remove the autoformat config

Hypfer reviewed (changes requested):
Just some remaining styling issues

enwi commented:
@Hypfer Happy now? 😄
